### PR TITLE
[transaction builders] Remove transaction builder shim

### PR DESCRIPTION
**Pull request by tzakian**
This removes the shim layer for transaction builders that we had for the
conversion from transaction scripts to script functions and adding
transaction builder support. The transaction builders under `client` are
now updated on a rebuild of the Diem Framework.

### CODE REVIEW

**Comment by libra-action** [comment]
**This PR may have modified JSON-RPC server code.**

Breaking changes policy:

1. Chaning JSON-RPC method signatures or removing/modifying fields
   in the response in /V1 will not be accepted.
2. Adding fields is ok.
3. Adding a new JSON-RPC method is ok.

Please ensure you have also done the following to update the docs:

1. Update json-rpc/API-CHANGELOG.md.
2. Add/update type/method documentation under json-rpc/docs.

File: client/transaction-builder/src/stdlib.rs [hunk 1]
```diff
@@ -106,6 +108,41 @@ pub enum ScriptCall {
...
+    /// | `Errors::INVALID_ARGUMENT` | `SlidingNonce::ENONCE_ALREADY_RECORDED`                                | The `sliding_nonce` has been previously recorded.                                          |
+    /// | `Errors::REQUIRES_ADDRESS` | `CoreAddresses::EDIEM_ROOT`                                           | The sending account is not the Diem Root account.                                         |
+    /// | `Errors::REQUIRES_ROLE`    | `Roles::EDIEM_ROOT`                                                   | The sending account is not the Diem Root account.                                         |
+    /// | `Errors::INVALID_ARGUMENT` | `DiemTransactionPublishingOption::EINVALID_SCRIPT_HASH`               | The script `hash` is an invalid length.                                                    |
+    /// | `Errors::INVALID_ARGUMENT` | `DiemTransactionPublishingOption::EALLOWLIST_ALREADY_CONTAINS_SCRIPT` | The on-chain allowlist already contains the script `hash`.                                 |
+    AddToScriptAllowList { hash: Bytes, sliding_nonce: u64 },
```
Comment (bmwill) [inline client/transaction-builder/src/stdlib.rs:144]: I thought @sblackshear removed the ability to modify the allowlist? Also I thought we wouldn't be allowing any additions to this list.

Comment (tzakian) [reply client/transaction-builder/src/stdlib.rs:144]: Ah, it's still in the legacy ABIs so gets generated here automatically, I'll remove it and regenerate.

**Review by moeziniaf — APPROVED** [review]

**Review by vgao1996 — APPROVED** [review]
LGTM!

**Pull request closed without merging**